### PR TITLE
AI upgrade logging/hints

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
tweak: AIs that get upgraded with a combat software upgrade now see a reminder that this doesn't change their laws or antagonist status.
admin: Added logging and an admin follow message to AI combat/surveillance software upgrades.
/:cl:

I've heard complaints about the AI combat upgrade: 
1) No logging
2) Players assume that this makes them an antag and start plasma flooding, which in return makes admins drink when they have to deal with this
3) Players don't know how to upgrade their harmyeller

This PR adds logging plus an admin alert to both upgrades, displays a little "hey, this upgrade doesn't make you an antag" hint and lets players know how to install the upgrade.

### CODE REVIEW

File: code/game/objects/items/robot/ai_upgrades.dm [hunk 1]
```diff
@@ -4,7 +4,7 @@
 //Malf Picker
 /obj/item/malf_upgrade
 	name = "combat software upgrade"
-	desc = "A highly illegal, highly dangerous upgrade for artificial intelligence units, granting them a variety of powers as well as the ability to hack APCs."
+	desc = "A highly illegal, highly dangerous upgrade for artificial intelligence units, granting them a variety of powers as well as the ability to hack APCs.<br> This upgrade is certified malfunction-free and must be applied directly to an active AI core."
```
Comment (MrDoomBringer) [inline code/game/objects/items/robot/ai_upgrades.dm]:
```suggestion
	desc = "A highly illegal, highly dangerous upgrade for artificial intelligence units, granting them a variety of powers as well as the ability to hack APCs.<br> This upgrade does not override any active laws, and must be applied directly to an active AI core."
```

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/game/objects/items/robot/ai_upgrades.dm [hunk 2]
```diff
@@ -18,15 +18,18 @@
...
+		message_admins("[ADMIN_LOOKUPFLW(user)] has upgraded [ADMIN_LOOKUPFLW(AI)] with a [src].")
 	to_chat(user, "<span class='notice'>You upgrade [AI]. [src] is consumed in the process.</span>")
 	qdel(src)
 
 
 //Lipreading
 /obj/item/surveillance_upgrade
 	name = "surveillance software upgrade"
-	desc = "A software package that will allow an artificial intelligence to 'hear' from its cameras via lip reading."
+	desc = "An illegal software package that will allow an artificial intelligence to 'hear' from its cameras via lip reading and hidden microphones."
```
Comment (SpaceManiac) [inline code/game/objects/items/robot/ai_upgrades.dm]: Spurious change? Anyways how could a software update physically add microphones?

File: code/game/objects/items/robot/ai_upgrades.dm [hunk 3]
```diff
@@ -4,7 +4,7 @@
 //Malf Picker
 /obj/item/malf_upgrade
 	name = "combat software upgrade"
-	desc = "A highly illegal, highly dangerous upgrade for artificial intelligence units, granting them a variety of powers as well as the ability to hack APCs."
+	desc = "A highly illegal, highly dangerous upgrade for artificial intelligence units, granting them a variety of powers as well as the ability to hack APCs.<br> This upgrade does not override any active laws, and must be applied directly to an active AI core."
```
Comment (SpaceManiac) [inline code/game/objects/items/robot/ai_upgrades.dm]: Space after `<br>` unnecessary (but should have no ill effects)

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/game/objects/items/robot/ai_upgrades.dm (hunk 2 above)
Comment (81Denton) [reply code/game/objects/items/robot/ai_upgrades.dm]: The malf upgrade mentions hidden microphones in its desciption, I figured I'd change this one for consistency. 
My best bet is <s>black magic</s> bluespace technology.

https://github.com/tgstation/tgstation/blob/740adc637708f9164217ea4d927004fc4772c473/code/modules/antagonists/traitor/equipment/Malf_Modules.dm#L849

File: code/game/objects/items/robot/ai_upgrades.dm (hunk 3 above)
Comment (81Denton) [reply code/game/objects/items/robot/ai_upgrades.dm]: Oh that's good to know, let me remove the space.

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**